lms/static/js/student_account/logistration_factory.js

```diff
@@ -0,0 +1,21 @@
+;(function (define) {
+    'use strict';
+    define(['jquery', 'js/student_account/views/AccessView'],
+        function($, AccessView) {
+
+        var container = $('#login-and-registration-container');
+
+        return function () {
```

Comment:
Nit: my preference is to pass a dict of options to the factory, rather than have it depend upon data attributes that happen to be set in the environment. This makes the factory easier to test in isolation, and also makes the code easier to understand. For one example, see how the Teams Mako template interacts with its factory:

https://github.com/edx/edx-platform/blob/master/lms/djangoapps/teams/static/teams/js/teams_tab_factory.js
https://github.com/edx/edx-platform/blob/master/lms/djangoapps/teams/templates/teams/teams.html

Some of our other views construct the configuration dict in the Python view, pass it to the Mako template, and then use ``json.dumps`` to write it out as a JavaScript object. For example:

https://github.com/edx/edx-platform/blob/master/lms/static/js/student_profile/views/learner_profile_factory.js
https://github.com/edx/edx-platform/blob/master/lms/templates/student_profile/learner_profile.html
https://github.com/edx/edx-platform/blob/master/lms/djangoapps/student_profile/views.py#L41

My preference is for the latter because it keeps all the configuration in one place in the Python code that can be easily tested.